presto-main/src/test/java/com/facebook/presto/operator/TestDriver.java

```diff
@@ -447,6 +476,30 @@ public void close()
         }
     }
 
+    private static class AlwaysBlockedMemoryRevokingTableScanOperator
+            extends TableScanOperator
+    {
+        public AlwaysBlockedMemoryRevokingTableScanOperator(
+                OperatorContext operatorContext,
+                PlanNodeId planNodeId,
+                PageSourceProvider pageSourceProvider,
+                List<Type> types,
+                Iterable<ColumnHandle> columns)
+        {
+            super(operatorContext, planNodeId, pageSourceProvider, types, columns);
+        }
+
+        @Override
+        public ListenableFuture<?> isBlocked()
```

Comment:
`IsBlocked()` is somewhat special method, i.e. `Driver` consults this _before_ calling other methods and, while doing so, it also checks revoking-related conditions. This tests currently does what it should, but wouldn't it be better to self-request memory revoking in some other method (eg `addInput`)?

Reply:
The problem with other methods is that they do not get called if isBlocked returns not-done future :/

Reply:
yeah that's the tricky part. that also makes this test pretty hacky (although does its job)

Reply:
ah.... obviously :)